src/commands/moderation/prefixCommands/functions/modifyVersion.ts

```diff
@@ -0,0 +1,166 @@
+            value: alias,
+        },
+        {
+            name: 'Enabled',
+            value: enabled ? 'Yes' : 'No',
+        },
+    ],
+    footer: { text: `Version ID: ${versionId}` },
+    color: Colors.Green,
+});
+
+const noModLogs = makeEmbed({
+    title: 'Prefix Commands - Modified Version - No Mod Log',
+    description: 'I can\'t find the mod logs channel. Please check the channel still exists.',
+    color: Colors.Red,
+});
+
+export async function handleModifyPrefixCommandVersion(interaction: ChatInputCommandInteraction<'cached'>) {
+    await interaction.deferReply({ ephemeral: true });
+
+    const conn = getConn();
+    if (!conn) {
+        await interaction.followUp({ embeds: [noConnEmbed], ephemeral: true });
+        return;
+    }
+
+    const version = interaction.options.getString('version')!;
+    const name = interaction.options.getString('name') || '';
+    const emoji = interaction.options.getString('emoji') || '';
+    const alias = interaction.options.getString('alias')?.toLowerCase() || '';
+    const enabled = interaction.options.getBoolean('is_enabled');
+    const moderator = interaction.user;
+
+    const nameRegex = /^[\w-]+$/;
+    if (name && !nameRegex.test(name)) {
+        await interaction.followUp({ embeds: [wrongFormatEmbed(name)], ephemeral: true });
+        return;
+    }
+    if (alias && !nameRegex.test(alias)) {
+        // eslint-disable-next-line no-await-in-loop
+        await interaction.followUp({ embeds: [wrongFormatEmbed(alias)], ephemeral: true });
+        return;
+    }
+    if (name) {
+        const foundVersion = await PrefixCommandVersion.findOne({
+            name: {
+                $ne: version,
+                $eq: name,
+            },
+        });
+        if (foundVersion || name.toLowerCase() === 'generic') {
+            await interaction.followUp({ embeds: [alreadyExistsEmbed(version, `${name} already exists as a version.`)], ephemeral: true });
+            return;
+        }
+    }
+    if (alias) {
+        const foundVersion = await PrefixCommandVersion.findOne({
+            name: { $ne: version },
+            alias,
+        });
+        if (foundVersion || alias === 'generic') {
+            await interaction.followUp({ embeds: [alreadyExistsEmbed(version, `${alias} already exists as a version alias.`)], ephemeral: true });
+            return;
+        }
+        const foundCommandName = await PrefixCommand.findOne({
+            $or: [
+                { name: alias },
+                { aliases: alias },
+            ],
+        });
+        if (foundCommandName) {
+            await interaction.followUp({ embeds: [alreadyExistsEmbed(version, `${alias} already exists as a command or command alias.`)], ephemeral: true });
+            return;
+        }
+    }
+
+    //Check if the mod logs channel exists
+    const modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
+    if (!modLogsChannel) {
```

Comment:
```suggestion
    let modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS);
    if (!modLogsChannel || !modLogsChannel.isTextBased()) {
        modLogsChannel = null;
```
Might be safer to actually check if the channel is a text channel? Especially if we want to offer the codebase to other projects. Then, if it's not a text channel set it to null to prevent accidental attempts to send into a non-text channel.